src/meltano/cli/add.py

```diff
@@ -95,25 +112,38 @@ def add(
         plugin_refs=plugin_refs, plugins_service=plugins_service
     )
     if not dependencies_met:
+        tracker.track_command_event(cli_tracking.STARTED)
+        tracker.track_command_event(cli_tracking.ABORTED)
         raise CliError(f"Failed to install plugin(s): {err}")
 
     add_service = ProjectAddService(project, plugins_service=plugins_service)
 
-    plugins: List[ProjectPlugin] = []
-    tracker = LegacyTracker(project)
+    plugins: list[ProjectPlugin] = []
+    legacy_tracker = LegacyTracker(project)
     for plugin in plugin_names:
-        plugins.append(
-            add_plugin(
-                project,
-                plugin_type,
-                plugin,
-                inherit_from=inherit_from,
-                variant=variant,
-                custom=flags["custom"],
-                add_service=add_service,
+        try:
+            plugins.append(
+                add_plugin(
+                    project,
+                    plugin_type,
+                    plugin,
+                    inherit_from=inherit_from,
+                    variant=variant,
+                    custom=flags["custom"],
+                    add_service=add_service,
+                )
             )
-        )
-        tracker.track_meltano_add(plugin_type=plugin_type, plugin_name=plugin)
+        except Exception:
```

Comment:
@pnadolny13 @aaronsteers this is the biggest gotcha, if one of the plugins is not known to meltano and the add command fails, we have no way to track what the requested plugin was. We do at least send along the PluginsContext for any plugins that we were able to resolve up until we hit the failure.

Reply:
Does that information come through in some other way? Such as through the info in the CLI context?

Reply:
For this and also for `install`, I think in a future iteration I'd like to separate this into a thinner 'cli started' with all or no plugins, but then a separate 'start' and 'end' for each plugin's addition or installation.

Related:

Do we already have something like an "ExceptionContext" for trace info, call stack, error messages, or similar? I'm happy to log that though and keep it out of this round.

Reply:
So, in a future iteration... somthing like:

`
cli - start
add tap-foo - start
install tap-foo start
install tap-foo complete
add tap-foo - complete
add target-foo - start
install target-foo - start
install target-foo - failed
add target-foo - failed
cli - failed
```

Reply:
The exception context is a great idea! We can easily get the current stack and exception info with `sys.exc_info` (or something like that - I'm on my phone and don't want to look it up). Then we can either send it with every event, or send it if and only if there is an exception currently being handled.